modules/formulize/class/templateScreen.php

```diff
@@ -0,0 +1,259 @@
+    const FORMULIZE_JS_FILE = "/modules/formulize/libraries/formulize.js";
+
+    function formulizeTemplateScreenHandler(&$db) {
+        $this->db =& $db;
+    }
+
+    function &getInstance(&$db) {
+        static $instance;
+        if (!isset($instance)) {
+            $instance = new formulizeTemplateScreenHandler($db);
+        }
+        return $instance;
+    }
+
+    function &create() {
+        return new formulizeTemplateScreen();
+    }
+
+    function insert($screen) {
+
+        $update = ($screen->getVar('sid') == 0) ? false : true;
+
+        if(!$sid = parent::insert($screen)) {
+            return false;
+        }
+        $screen->assignVar('sid', $sid);
+
+        if (!$update) {
+            $sql = sprintf("INSERT INTO %s (sid, custom_code, template) VALUES (%u, %s, %s)", $this->db->prefix('formulize_screen_template'),
+                $screen->getVar('sid'), $this->db->quoteString($screen->getVar('custom_code')), $this->db->quoteString($screen->getVar('template')));
+        } else {
+            $sql = sprintf("UPDATE %s SET custom_code = %s, template = %s WHERE sid = %u", $this->db->prefix('formulize_screen_template'),
+                $this->db->quoteString($screen->getVar('custom_code')), $this->db->quoteString($screen->getVar('template')), $sid);
+        }
+        $result = $this->db->query($sql);
+        if (!$result) {
+            print "Error: could not save template screen properly: ".$this->db->error()." for query: $sql";
+            return false;
+        }
+
+        $success1 = true;
+        if(isset($_POST['screens-custom_code'])) {
+            $success1 = $this->write_custom_code_to_file(trim($_POST['screens-custom_code']), $screen);
+        }
+        $success2 = true;
+        if(isset($_POST['screens-template'])) {
+            $success2 = $this->write_template_to_file(trim($_POST['screens-template']), $screen);
+        }
+        if (!$success1 || !$success2) {
+            return false;
+        }
+
+        return $sid;
+    }
+
+
+    function get($sid) {
+        $sid = intval($sid);
+        if ($sid > 0) {
+            $sql = 'SELECT * FROM '.$this->db->prefix('formulize_screen').' AS t1, '. $this->db->prefix('formulize_screen_template').' AS t2 WHERE t1.sid='.$sid.' AND t1.sid=t2.sid';
+            if (!$result = $this->db->query($sql)) {
+                return false;
+            }
+            $numrows = $this->db->getRowsNum($result);
+            if ($numrows == 1) {
+                $screen = new formulizeTemplateScreen();
+                $screen->assignVars($this->db->fetchArray($result));
+                return $screen;
+            }
+        }
+        return false;
+    }
+
+
+    function render($screen) {
+        global $xoTheme;
+        if($xoTheme) {
+            $xoTheme->addStylesheet(self::FORMULIZE_CSS_FILE);
+            $xoTheme->addScript(self::FORMULIZE_JS_FILE);
+        }
+
+        $custom_code_filename = custom_code_filename($screen);

```

Comment:
The custom_code_filename(), template_filename(), and run_template_php_code() functions are all members of the class, so they should be called like $this->function().